app/validators/special_character_validator.rb

```diff
@@ -0,0 +1,15 @@
+class SpecialCharacterValidator < ActiveModel::EachValidator
+
+  def validate_each(record, attribute, value)
+    return if value.nil?
+
+    if attribute == :address_building
```

Comment:
It's not good practice to make decisions based on which attribute is using it.  If a particular attribute needs something slightly different, it is either a different validator - OR - passing options into the validator such as

```ruby
validates :address_building, special_character: { option1: true, option2: false, any_option: 4 }
```

these options are passed in to the initialize method of the validator